### DISCUSSION TITLE
Is it possible to set security headers for Argo CD  from helm-chart?

### DISCUSSION THREAD

**Discussion by krikeri** [argoproj/argo-helm#2153] [Q&A]
Has anyone been able to set the following security headers for Argo CD server from helm-chart / config?

Headers
- Strict-Transport-Security 
- X-Content-Type-Options 
- Cache-control Header 
- X-XSS-Protection

Thank you for your answers!

**Comment by yu-croco** [comment]
Hi @krikeri , it depends on how Argo CD itself accepts these configuration. 
So please investigate the feature of Argo CD at first. 
*We manage Helm Chart but not Argo CD itself here.

**Reply by mkilchhofer** [reply]
It looks like @krikeri found the upstream issue for that:
- https://github.com/argoproj/argo-cd/issues/7778

Another option would be to expose Argo CD via an ingress controller and set these headers via the ingress annotations. Eg. on ingress-nginx:
https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#configuration-snippet
